sqlglot/optimizer/qualify_columns.py

```diff
@@ -413,7 +445,12 @@ def _expand_stars(
             if not pivot_output_columns:
                 pivot_output_columns = [c.alias_or_name for c in pivot.expressions]
 
+    scope_annotated = False
+    scope_cols = scope.columns
+
     for expression in scope.expression.selects:
+        tables = []
```

Comment:
This seems like dead code

Reply:
Hm I guess not actually, but let's not initialize this here. Instead, move it to the new `elif` branch.

Reply:
Sorry, can you specify which would that be?

Reply:
Yep, we can move this to L508.